src/cloudant/document.py

```diff
@@ -351,5 +401,5 @@ def put_attachment(self, attachment, content_type, data, headers=None):
             headers=headers
         )
         resp.raise_for_status()
-
+        self.fetch()
```

Comment:
I made the necessary change to the `delete_attachment` method.  Done in fcc5059.